src/linalg/cholUnblocked.jl

```diff
@@ -9,24 +9,29 @@ because these are part of the inner calculations in a blocked Cholesky factoriza
 function cholUnblocked! end
 
 function cholUnblocked!(D::Diagonal{T}, ::Type{Val{:L}}) where {T<:AbstractFloat}
-    map!(sqrt, D.diag, D.diag)
+    Ddiag = D.diag
+    @inbounds for i in eachindex(Ddiag)
+        (ddi = Ddiag[i]) ≤ zero(T) && throw(LinearAlgebra.PosDefException(i))
+        Ddiag[i] = sqrt(ddi)
+    end
+
     return D
 end
 
 function cholUnblocked!(A::StridedMatrix{T}, ::Type{Val{:L}}) where {T<:BlasFloat}
     n = LinearAlgebra.checksquare(A)
     if n == 1
-        A[1] < zero(T) && throw(PosDefException(1))
+        A[1] < zero(T) && throw(LinearAlgebra.PosDefException(1))
         A[1] = sqrt(A[1])
     elseif n == 2
-        A[1] < zero(T) && throw(PosDefException(1))
+        A[1] < zero(T) && throw(LinearAlgebra.PosDefException(1))
```

Comment:
why do we need this module qualification now?

Reply:
I had forgotten that `PosDefException` was listed in a `using` statement.  I will revert to the unqualified name in all the uses in this file.